pkg/networkservice/core/adapters/server_to_client.go

```diff
@@ -44,9 +44,14 @@ func (s *serverToClient) Request(ctx context.Context, in *networkservice.Network
 		return nil, err
 	}
 	if in == nil {
-		in = &networkservice.NetworkServiceRequest{}
+		in = &networkservice.NetworkServiceRequest{
+			Connection: &networkservice.Connection{},
+		}
+	}
+	if conn != nil {
+		in.Connection.Context = conn.Context
+		in.Connection.Path = conn.Path
```

Comment:
@haiodo What do you think about this simple patch?
```suggestion
		in = &networkservice.NetworkServiceRequest{}
	}
	if conn != nil {
	    if in.Connection == nil {
	        in.Connection = conn
	    } else {
		in.Connection.Context = conn.Context
		in.Connection.Path = conn.Path
	     }
```